src/test/java/uk/gov/hmcts/reform/bulkscanprocessor/services/UploadEnvelopeDocumentsServiceTest.java

```diff
@@ -0,0 +1,273 @@
+            .createEvent(eventCaptor.capture(), eq(CONTAINER_1), eq(ZIP_FILE_NAME), eq("failed"), eq(null));
+        assertThat(eventCaptor.getValue()).isEqualTo(Event.DOC_UPLOAD_FAILURE);
+    }
+
+    @Test
+    void should_mark_as_doc_upload_failure_when_unable_to_upload_pdfs()
+        throws URISyntaxException, StorageException, IOException {
+        // given
+        given(blobManager.getContainer(CONTAINER_1)).willReturn(blobContainer);
+        given(blobContainer.getName()).willReturn(CONTAINER_1);
+        given(blobContainer.getBlockBlobReference(ZIP_FILE_NAME)).willReturn(blockBlob);
+        given(blobManager.acquireLease(blockBlob, CONTAINER_1, ZIP_FILE_NAME)).willReturn(Optional.of(LEASE_ID));
+        given(blockBlob.openInputStream()).willReturn(blobInputStream);
+        given(zipFileProcessor.process(any(ZipInputStream.class), eq(ZIP_FILE_NAME)))
+            .willReturn(new ZipFileProcessingResult(new byte[]{}, emptyList())); // unit test doesn't care if it's empty
+
+        // and
+        willThrow(new RuntimeException("oh no")).given(documentProcessor).uploadPdfFiles(emptyList(), emptyList());
+
+        // and
+        List<Envelope> envelopes = getEnvelopes();
+
+        // when
+        uploadService.processByContainer(CONTAINER_1, envelopes);
+
+        // then
+        ArgumentCaptor<Event> eventCaptor = ArgumentCaptor.forClass(Event.class);
+        verify(envelopeProcessor, times(1))
+            .createEvent(eventCaptor.capture(), eq(CONTAINER_1), eq(ZIP_FILE_NAME), eq("oh no"), eq(null));
+        assertThat(eventCaptor.getValue()).isEqualTo(Event.DOC_UPLOAD_FAILURE);
+
+        // and
+        verify(envelopeProcessor, times(1)).markAsUploadFailure(envelopes.get(0));
+    }
+
+    @Test
+    void should_mark_as_uploaded_when_everything_went_well()
+        throws URISyntaxException, StorageException, IOException {
+        // given
+        given(blobManager.getContainer(CONTAINER_1)).willReturn(blobContainer);
+        given(blobContainer.getName()).willReturn(CONTAINER_1);
+        given(blobContainer.getBlockBlobReference(ZIP_FILE_NAME)).willReturn(blockBlob);
+        given(blobManager.acquireLease(blockBlob, CONTAINER_1, ZIP_FILE_NAME)).willReturn(Optional.of(LEASE_ID));
+        given(blockBlob.openInputStream()).willReturn(blobInputStream);
+        given(zipFileProcessor.process(any(ZipInputStream.class), eq(ZIP_FILE_NAME)))
+            .willReturn(new ZipFileProcessingResult(new byte[]{}, emptyList())); // unit test doesn't care if it's empty
+        willDoNothing().given(documentProcessor).uploadPdfFiles(emptyList(), emptyList());
+
+        // when
+        uploadService.processByContainer(CONTAINER_1, getEnvelopes());
+
+        // then
+        ArgumentCaptor<Event> eventCaptor = ArgumentCaptor.forClass(Event.class);
+        verify(envelopeProcessor, times(1)).handleEvent(any(Envelope.class), eventCaptor.capture());
+        assertThat(eventCaptor.getValue()).isEqualTo(Event.DOC_UPLOADED);
```

Comment:
also that it uploaded documents from zip?

Reply:
that's mock definition in `given` step:

```java
willDoNothing().given(documentProcessor).uploadPdfFiles(emptyList(), emptyList());
```

if it wasn't used but stubbings defined in test, you'll get an error:

```
Unnecessary stubbings detected.
Clean & maintainable test code requires zero unnecessary code.
Following stubbings are unnecessary (click to navigate to relevant line of code):
  1. -> at uk.gov.hmcts.reform.bulkscanprocessor.services.UploadEnvelopeDocumentsServiceTest.should_mark_as_uploaded_when_everything_went_well(UploadEnvelopeDocumentsServiceTest.java:243)
```

tl;dr - there's no point in verifying what you stubbed

Reply:
From readability point of view, I'd rather be able to see what the tested method is supposed to do in the 'then' stage, not in the 'given' stage.

No need to stub.

Reply:
for that method call it can be converted to verification step instead of stub. 1s